crates/extensions/c8y_mapper_ext/src/tests.rs

```diff
@@ -2527,85 +2529,127 @@ async fn handle_log_upload_executing_and_failed_cmd_for_child_device() {
 #[tokio::test]
 async fn handle_log_upload_successful_cmd_for_main_device() {
     let ttd = TempTedgeDir::new();
-    let (mqtt, http, _fs, _timer, _dl) = spawn_c8y_mapper_actor(&ttd, true).await;
+    let (mqtt, http, _fs, _timer, ul, _dl) = spawn_c8y_mapper_actor(&ttd, true).await;
     spawn_dummy_c8y_http_proxy(http);
 
     let mut mqtt = mqtt.with_timeout(TEST_TIMEOUT_MS);
+    let mut ul = ul.with_timeout(TEST_TIMEOUT_MS);
     skip_init_messages(&mut mqtt).await;
 
     // Simulate a log file is uploaded to the file transfer repository
-    ttd.dir("tedge")
+    let uploaded_file = ttd
         .dir("file-transfer")
-        .dir("main")
+        .dir("test-device")
         .dir("log_upload")
         .file("typeA-1234");
 
-    // Simulate log_upload command with "executing" state
+    // Simulate log_upload command with "successful" state
     mqtt.send(MqttMessage::new(
         &Topic::new_unchecked("te/device/main///cmd/log_upload/1234"),
         json!({
             "status": "successful",
-            "tedgeUrl": format!("http://localhost:8888/tedge/file-transfer/main/log_upload/typeA-1234"),
+            "tedgeUrl": "http://localhost:8888/tedge/file-transfer/main/log_upload/typeA-1234",
```

Comment:
This `tedgeUrl` still uses `main` in the path, when `main` was replaced with `test-device` in the `uploaded_file` path above.